packages/medusa/src/services/order-edit.ts

```diff
@@ -312,4 +313,75 @@ export default class OrderEditService extends TransactionBaseService {
       await orderEditRepo.remove(edit)
     })
   }
+
+  async decline(
+    orderEditId: string,
+    declinedReason: string | undefined,
+    customerId?: string
+  ): Promise<OrderEdit> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEditRepo = manager.getCustomRepository(
+        this.orderEditRepository_
+      )
+
+      const orderEdit = await this.retrieve(orderEditId)
+
+      if (orderEdit.status === OrderEditStatus.DECLINED) {
+        throw new MedusaError(
```

Comment:
suggestion: Here if we make it idempotent as mention in a previous comment, we could just return the order edit